src/surfing_penguin/routes.py

```diff
@@ -1,68 +1,54 @@
 """routes.py: Each function in this file indicates a web page (HTML page)."""
 
-from surfing_penguin import surfing_penguin, api
-from flask import render_template, flash, redirect, url_for
+from surfing_penguin import surfing_penguin, api, login_manager
 from flask_restplus import Resource, fields
-from surfing_penguin.forms import LoginForm
-from surfing_penguin.db_interface import Qstnr, UserFunctions
-
-question = api.model("question_model", {
-        'content': fields.String,
-        'id': fields.Integer
-    })
+from surfing_penguin.db_interface import UserFunctions
+from flask_login import login_user, logout_user, current_user, login_required
 
 
+api_return_message = api.model("return_message_model", {
+        'messages': fields.String
+    })
 api_get_user = api.model("get_user_model", {
         'username': fields.String,
-        'id': fields.Integer,
         'password': fields.String,
     })
 
 
 api_return_user = api.model("return_user_model", {
         'username': fields.String,
-        'id': fields.Integer,
         'messages': fields.String,
     })
 
 
 api_show_user = api.model("show_user_model", {
         'username': fields.String,
-        'id': fields.Integer,
     })
 
 
-qstnrs = [
-    {
-        'author': {'username': 'John'},
-        'title': 'QSTQ',
-        'id': 0
-    },
-    {
-        'author': {'username': 'Susan'},
-        'title': 'QSTQQ',
-        'id': 1
-    }
-]
-qstnr1 = Qstnr()
-qstnr1.new_qst({'content': 'Q1'})
-qstnr1.new_qst({'content': 'Q2'})
-
-
-@api.route('/api_display')
-class api_display(Resource):
-    """ This api is for developers to monitor the program,
-        currently only showing the questionnaire.
-        usage: start the server, and curl localhost:port/api_display """
-    @api.marshal_list_with(question)
-    def get(self):
-        return qstnr1.questions
-
-
+api_show_user_and_time = api.model("show_user_and_time_model", {
+        'username': fields.String,
+        'last_seen': fields.DateTime
+    })
 """ user account associated APIs:
     register, login, show_users, delete_user, search_user """
 
 
+@surfing_penguin.before_request
+def before_request():
```

Comment:
It would be better to add comments for this function (and other not self-explained ones).